File: CSE_Directory/CSE_Assignments/CSE_3302-PL/lab01/lxm_lab01.java
/* Landon Moon
 * 1001906270
 * java 17.0.2
 * Windows 10 / Omega
 */

 import java.io.File;


public class lxm_lab01 {
  public static int dirSize(File dir) {
    int total = 0;

    // loop through files/directories
    for( File f: dir.listFiles() )
    {
      // if dir
      if(f.isDirectory())
        total += dirSize(f);
      // if file
      else
        total += f.length();
    }

    return total;
  }

  public static void main(String args[]) {
    // start at current dir
    int size = dirSize(new File("."));
    
    System.out.println(size);
  }
}

File: CSE_Directory/CSE_Assignments/archive/CSE_3302-PL/lab01/lxm_lab01.py
# Landon Moon
# 1001906270
# python 3.10.9
# Windows 10 / Omega

import os

def dirSize(path):
  total=0

  # loop through all files
  for f in os.listdir(path):
    newPath = path + '/' + f

    # if dir
    if os.path.isdir(newPath):
      total+=dirSize(newPath)
    # if file
    else:
      total+=os.path.getsize(newPath)

  return total

# start in current directory
size = dirSize(".")
print(size)
